src/analyses/variable-sensitivity/abstract_enviroment.h

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************\
+
+ Module: analyses variable-sensitivity
+
+ Author: Thomas Kiley, thomas.kiley@diffblue.com
+
+\*******************************************************************/
+#ifndef CPROVER_ANALYSES_VARIABLE_SENSITIVITY_ABSTRACT_ENVIROMENT_H
+#define CPROVER_ANALYSES_VARIABLE_SENSITIVITY_ABSTRACT_ENVIROMENT_H
+
+#include <map>
+#include <memory>
+
+#include <util/std_expr.h>
+#include <util/message.h>
+#include <analyses/variable-sensitivity/abstract_object.h>
+
+class abstract_environmentt
+{
+public:
+  // These three are really the heart of the method
+  virtual abstract_object_pointert eval(
+    const exprt &expr, const namespacet &ns) const;
+  virtual bool assign(const exprt &expr, const abstract_object_pointert value);
+  virtual bool assume(const exprt &expr, const namespacet &ns);
+
+  virtual abstract_object_pointert abstract_object_factory(
+    const typet type, bool top=true, bool bottom=false) const;
+  // For converting constants in the program
+  // Maybe these two should be compacted to one call...
+  virtual abstract_object_pointert abstract_object_factory(
+    const typet type, const constant_exprt e) const;
+
+
+  virtual bool merge(const abstract_environmentt &env);
+
+  // This should be used as a default case / everything else has failed
+  // The string is so that I can easily find and diagnose cases where this occurs
+  virtual void havoc(const std::string &havoc_string);
```

Comment:
This seems to have a single use case. I'm not sure that's a good code/value ratio.